apis/Google.Devtools.AspNet/Google.Devtools.AspNet.IntegrationTests/CloudErrorReporting/CloudErrorReportingExceptionLoggerIntregrationTest.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using Google.Devtools.Clouderrorreporting.V1Beta1;
+using Microsoft.Owin.Testing;
+using Owin;
+using System;
+using System.Globalization;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Http;
+using System.Web.Http.ExceptionHandling;
+using Xunit;
+
+namespace Google.Devtools.AspNet.IntegrationTests
+{
+    public class CloudErrorReportingExceptionLoggerIntregrationTest
+    {
+        private const string ProjectEnvironmentVariable = "TEST_PROJECT";
+        private static string ProjectId;
+        private static string TestId;
+
+        public CloudErrorReportingExceptionLoggerIntregrationTest()
+        {
+            ProjectId = Environment.GetEnvironmentVariable(ProjectEnvironmentVariable);
+            if (string.IsNullOrEmpty(ProjectId))
+            {
+                throw new InvalidOperationException(
+                    $"Please set the {ProjectEnvironmentVariable} environment variable before running tests");
+            }
+            TestId = DateTime.UtcNow.ToString("'test'_yyyyMMddTHHmmssFFF", CultureInfo.InvariantCulture);
+        }
+
+        [Fact]
+        public async void ErrorReportingTest()
+        {
+            Task<ErrorStatsServiceClient> clientTask = ErrorStatsServiceClient.CreateAsync();
+
+            // Create a test server and make an http.
+            TestServer server = TestServer.Create<ErrorReportingApplication>();
+            await server.HttpClient.GetAsync("");
+
+            // Create a request that will filter on the TestId which is set to the service and version.
+            ListGroupStatsRequest request = new ListGroupStatsRequest
+            {
+                ProjectName = ErrorStatsServiceClient.FormatProjectName(ProjectId),
+                ServiceFilter = new ServiceContextFilter
+                {
+                    Service = TestId,
+                    Version = TestId
+                }
+            };
+
+            // Check that we have the proper results and the TestId shows up in the error.
+            ErrorStatsService.ErrorStatsServiceClient grpcClient = (await clientTask).GrpcClient;
```

Comment:
Any reason you're not using `client.ListGroupStats(...)`? Is it because we're not exposing a version that accepts the filter? If so, please file a bug and we'll sort that out. Any time you have to drop down to the gRPC client directly, we should find out why.